config/squidGuard/squidguard_blacklist.php

```diff
@@ -58,11 +58,19 @@
 function squidguard_blacklist_AJAX_response( $request )
 {
     $res = '';
+    $status = '';
     $sz  = 0;
     $pcaption = '&nbsp;';
 
     # Actions
-    if     ($request['blacklist_download_start'])  squidguard_blacklist_update_start( $request['blacklist_url'] ); # update start
+    if ($request['blacklist_download_start'])
+        {
+                $url = $request['blacklist_url'];
+                if ( preg_match('/["\'\s\n\r\(\);]+/', $url )){
+                        $status = "Error: Bad blacklist URL";
+                        squidguard_update_log($status,"");
+                }else   squidguard_blacklist_update_start( $url ); # update start
```

Comment:
See the style guide for recommendations here -- always use curly braces in these cases. The code in that file is a bit of a mess but it's still better to make things right as it's edited.